src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ParameterBuilderImpl.cs

```diff
@@ -30,7 +30,12 @@ public ParameterBuilderImpl(MethodBuilderImpl methodBuilder, int sequence, Param
 
         public override int Position => _position;
 
-        public override void SetConstant(object? defaultValue) => _defaultValue = defaultValue;
+        public override void SetConstant(object? defaultValue)
+        {
+            FieldBuilderImpl.ValidateDefaultValueType(defaultValue, _methodBuilder.ParameterTypes![_position - 1]);
```

Comment:
Is checking the length here appropriate? The previous version of this didn't throw and this one now does/can.

Reply:
Yes, we should add length check.